src/components/MovieDetails/MovieDetails.js

```diff
@@ -25,6 +25,21 @@ class MovieDetails extends Component {
       .then(data => this.props.fetchUserRatings(data.ratings))
   }
 
+  removeRating = (e) => {
+    e.preventDefault();
+    const matchingMovie = this.props.userRatings.find(rating => rating.movie_id === this.props.id);
+    if(matchingMovie) {
+      const deleteLink = `https://rancid-tomatillos.herokuapp.com/api/v1/users/${this.props.userInfo.id}/ratings/${matchingMovie.id}`
+      fetch(deleteLink, {
+        method: "DELETE"
+      })
+      .then(() => apiFetchRatings(this.props.userInfo.id))
+      .then(data => this.props.fetchUserRatings(data.ratings))
+    } else if (!matchingMovie) {
+      console.log("not rated yet")
+    }
```

Comment:
Should it even be possible to get to this point? Should the delete button just be disabled/invisible if the movie is unrated?

Reply:
I added that as a default for error.  I think code can break if there is no error handling

Reply:
I agree with Cristina, I think the user won't even be able to click the remove rating button if it is not showing up. So the function should just be able to work without the if statement without breaking?

Reply:
It looks like the `toggleRatingButtons()` is already doing that validation

Reply:
Ok. I was using the console.log for a previous version of that code and I had changed the code in the final version.  So, you are right, the console.log is not needed. I was using the console.log to test the initial function I had started. Good catch!

Reply:
Cristina and I were referring to the if (matchMovie) statement on line 31. The only time that Remove Rating button shows up is if the user has rated the movie (this logic is in line 66). So I don't think you need to run that if statement again on line 31.

Reply:
I made the changes from our checkin.